clients/src/main/java/org/apache/kafka/clients/consumer/internals/MembershipManagerImpl.java

```diff
@@ -1338,7 +1339,14 @@ private CompletableFuture<Void> enqueueConsumerRebalanceListenerCallback(Consume
                                                                              Set<TopicPartition> partitions) {
         SortedSet<TopicPartition> sortedPartitions = new TreeSet<>(TOPIC_PARTITION_COMPARATOR);
         sortedPartitions.addAll(partitions);
-        CompletableBackgroundEvent<Void> event = new ConsumerRebalanceListenerCallbackNeededEvent(methodName, sortedPartitions);
+
+        // We don't yet have the concept of having an expiring callback, but we will likely want that eventually.
+        Timer timer = time.timer(Long.MAX_VALUE);
+        CompletableBackgroundEvent<Void> event = new ConsumerRebalanceListenerCallbackNeededEvent(
+            methodName,
+            sortedPartitions,
+            timer
+        );
```

Comment:
Here I have a similar comment as with `AsyncCommitEvent`, I would move the creation of the timer into the constructor of `ConsumerRebalanceListenerCallbackNeededEvent`.

Reply:
I'll look into how to do this in a way that I don't find too ugly 😉

Reply:
I was able to refactor the code and this is now sans `Timer` again.